### PR TITLE
Restore DILL_AVAILABLE for backwards compat with torchdata

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122616

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @albanD

### CODE REVIEW

File: torch/utils/data/datapipes/utils/common.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from functools import partial
 from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Union
 from torch.utils._import_utils import dill_available
+from torch.testing._internal.common_utils import LazyVal
```
Comment (peterbell10) [inline torch/utils/data/datapipes/utils/common.py:12]: Is it kosher to import internal testing utils here? `dill_available` doesn't actually import `dill` so I think it's fine to call it at import time.

**Review by peterbell10 — APPROVED** [review]

**Comment by Rhett-Ying** [comment]
Is there any workaround for this issue if working with torch 2.3.0?

**Comment by albanD** [comment]
I would expect that you can work around this by doing the imports this way, not ideal though:
```python
import torch
torch.utils.data.datapipes.utils.common.DILL_AVAILABLE = torch.utils._import_utils.dill_available()
import torchdata
```

**Comment by Atul-1992** [comment]
Thanks, It worked!

**Comment by sarikasaxena30** [comment]
FileNotFoundError: Cannot find DGL C++ graphbolt library at /usr/local/lib/python3.10/dist-packages/dgl/graphbolt/libgraphbolt_pytorch_2.3.0.so
How to solve this problem?anyone can help me in this

**Pull request closed without merging**